scripts/py2md.py

```diff
@@ -223,10 +230,56 @@ def main():
             render_module_header=False,
         ),
     )
+    if out:
+        out_path = Path(out)
+        os.makedirs(out_path.parent, exist_ok=True)
 
     pydocmd.load_modules()
     pydocmd.process()
     pydocmd.render()
+    logging.info("Processed %s", module)
+
+
+def _py2md_wrapper(x: Tuple[str, str]):
+    """
+    Used to wrap py2md since we can't pickle a lambda (needed for multiprocessing).
+    """
+    return py2md(x[0], x[1])
+
+
+def parse_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("modules", nargs="+", type=str, help="""The Python modules to parse.""")
+    parser.add_argument(
+        "-c",
+        "--concurrency",
+        type=int,
+        default=1,
+        help="""The number of threads to use. Defaults to 1.""",
+    )
+    parser.add_argument(
+        "-o",
+        "--out",
+        nargs="+",
+        type=str,
+        help="""Output files.
+                If given, must have the same number of items as 'modules'.
+                If not given, stdout is used.""",
+    )
+    return parser.parse_args()
+
+
+def main():
+    opts = parse_args()
+    outputs = opts.out if opts.out else [None] * len(opts.modules)
+    if opts.concurrency > 1 and opts.out:
+        logging.info("Using %d threads", opts.concurrency)
+        with Pool(opts.concurrency) as p:
+            list(p.imap(_py2md_wrapper, zip(opts.modules, outputs), 10))
```

Comment:
Apparently the idiomatic way of exhausting an iterator is `collections.deque(iterator, maxlen=0)`?

Reply:
Ahhhhhh thank you!